frontend/src/shared/util/table-utils.tsx

```diff
@@ -238,7 +238,7 @@ export const ServerSidePaginator = <T extends ServerRequestProps>(props: ServerS
     const loading = props.loading;
     const setLoading = props.setLoading;
 
-    const fetchData = async (usePagination: boolean, callback?: () => void) => {
+    const fetchData = useCallback(async (usePagination: boolean, callback?: () => void) => {
```

Comment:
why do we need to memoize this function? seems like it makes no difference to memoize it since its dependent on so many things that would change anyways when we want to call this.

Reply:
I think you're right we can leave this non memoized.